### PR TITLE
add proxy support

### PR DESCRIPTION
**Pull request by antonioua**
Add proxy support for http client

### CODE REVIEW

**Comment by vans163** [comment]
Does this support username+password? Also this is HTTP proxy only?

**Comment by pasha-i** [comment]
@antonioua can we use SOCKS5 proxy as well? not exactly sure if used http client would work with them

**Comment by antonioua** [comment]
yep, tested socks5 and http - both works.

**Comment by pasha-i** [comment]
saw another PR https://github.com/codesenberg/bombardier/compare/master...mariotrucco:78-add-proxy-support for proxy support, that one assumes it can only be a http proxy, but otherwise looks reasonable as well

**Comment by vans163** [comment]
> this proxy is for http client only, doesn't support username/pass but I can try to add it @vans163 .

Yea I think its important to have user/pass auth for both http and socks5 proxy as most paid proxies use that. Free/open/tor proxies dont but those are very few.

File: bombardier.go [hunk 1]
```diff
@@ -441,6 +441,8 @@ func (b *bombardier) disableOutput() {
 }
 
 func main() {
+	proxyServer = os.Getenv("BOMBARDIER_PROXY")
```
Comment (sebastienros) [inline bombardier.go:444]: Should probably come from a command line argument

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM golang:1.15 as builder
```
Comment (sebastienros) [inline Dockerfile:1]: Shouldn't this file be in a different PR?

Comment (antonioua) [reply Dockerfile:1]: I'll remove Dockerfile, was just curious if you need it.

File: bombardier.go (hunk 1 above)
Comment (antonioua) [reply bombardier.go:445]: done

**Comment by antonioua** [comment]
added proxy http/socks5 support for both http and fasthttp clients
